instrument/src/Instrument/Client.hs

```diff
@@ -219,6 +223,31 @@ timeI nm hostDimPolicy rawDims i act = do
   submitTime nm hostDimPolicy rawDims secs i
   return res
 
+-- | Run a monadic action while measuring its runtime. Push the measurement into
+-- the instrument system. rethrows exceptions and sends a different Metric on
+-- failure
+--
+-- >>> timeExI \"fileUploadTimeError\" \"fileUploadTime\" policy dims instr $ uploadFile file
+timeExI
+  :: (MonadIO m, MonadCatch m)
+  => MetricName
```

Comment:
I wonder if we should make this API a bit more flexible. It seems to me that if there's an error, in addition to using a different metric name, we might want to have a different host dimension policy and set of dimensions. What if we changed it so the type was `(Either SomeException a -> (MetricName, HostDimensionPolicy, Dimensions)) -> Instrument -> m a -> m a`?

Reply:
Yes, at the very least it doesn't confuse the caller which one is which. updated.